threads/threads.ts

```diff
@@ -668,6 +682,11 @@ export async function run<ReturnType>(task: (() => ReturnType)|JSTransferableFun
 			const variableName = e.message.match(/ReferenceError - (\S*)/)![1];
 			throw new RuntimeError("Variable '"+variableName+"' from the parent scope must be explicitly declared at the beginning of the function body with 'use ("+variableName+")'.")
 		}
+		else if (e instanceof Error && e.message.endsWith("is offline")) {
```

Comment:
This is more a question to think about in general, not just for this occurrence in this PR.

Reply:
Yes, normally we would get a `NetworkError` here, but for some reason it was cast to a normal `Error` so I just did this check as a workaround for now. This can definitely be improved.